src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Runtime/ClientServicesConfig.cs

```diff
@@ -0,0 +1,54 @@
+﻿// ------------------------------------------------------------
+//  Copyright (c) Microsoft Corporation.  All rights reserved.
+//  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+// ------------------------------------------------------------
+
+namespace Microsoft.Azure.IIoT.OpcUa.Protocol.Runtime {
+    using Microsoft.Azure.IIoT.Utils;
+    using Microsoft.Extensions.Configuration;
+    using System;
+    /// <summary>
+    /// Client's application configuration implementation
+    /// </summary>
+    public class ClientServicesConfig : ConfigBase, IClientServicesConfig {
+
+        private const string kPkiRootFolder = "PkiRootFolder";
+        private const string kApplicationCertificateFolder = "ApplicationCertificateFolder";
+        private const string kTrustedPeerCertificatesFolder = "TrustedPeerCertificatesFolder";
+        private const string kTrustedIssuerCertificatesFolder = "TrustedIssuerCertificatesFolder";
+        private const string kRejectedCertificatesFolder = "RejectedCertificatesFolder";
+        private const string kAutoAcceptUntrustedCertificates = "AutoAcceptUntrustedCertificates";
+
+        /// <inheritdoc/>
+        public string PkiRootFolder =>
+            GetStringOrDefault(kPkiRootFolder, null);
```

Comment:
Would be good to also add flat environment variable names for the pki root folder so that they can be set in the simulation.   Furthermore, would be great to prefix the configuration properties with something like "ClientServices::" to scope them.

Reply:
Oh - and here we could also set reasonable default values if that  makes sense, i.e. for at least anything other than the root

Reply:
* the pki folder acts as a global setting when the other folders are not defined. 
* all the values are defaulted now here. Previously they were defaulted in the initialization function
* I need to figure out how to add environment variables. For simulation, we can use arguments to setup the twin
* I'm not sure what you mean with prefixed configuration properties. do you mean the lookup keys constants?